infra/fastly/terraform/dl.k8s.io/services.tf

```diff
@@ -0,0 +1,89 @@
+resource "fastly_service_vcl" "files" {
+  name     = "cdn.dl.k8s.io"
+  activate = true
+
+  domain {
+    name = "cdn.dl.k8s.io"
+  }
+
+  backend {
+    name             = "GCS"
+    auto_loadbalance = false
+    shield           = "bfi-wa-us"
+
+    healthcheck = "GCS Health"
+
+    address           = "storage.googleapis.com"
+    port              = 443
+    use_ssl           = true
+    ssl_cert_hostname = "storage.googleapis.com"
+    ssl_sni_hostname  = "storage.googleapis.com"
+
+    override_host = "${var.bucket}.storage.googleapis.com"
+
+    connect_timeout       = 5000
+    first_byte_timeout    = 60000
+    between_bytes_timeout = 15000
+    error_threshold       = 5
+  }
+
+  healthcheck {
+    name = "GCS Health"
+
+    host           = "${var.bucket}.storage.googleapis.com"
+    method         = "GET"
+    path           = "/"
+    check_interval = 3000
+    timeout        = 2000
+    threshold      = 2
+    initial        = 2
+    window         = 4
+  }
+
+  condition {
+    name      = "NeverReq"
+    type      = "REQUEST"
+    statement = "req.http.Fastly-Client-IP == \"127.0.0.1\" && req.http.Fastly-Client-IP != \"127.0.0.1\""
+  }
+
+  condition {
+    name      = "5xx Error"
+    type      = "RESPONSE"
+    statement = "(resp.status >= 500 && resp.status < 600)"
+  }
+
+  condition {
+    name      = "Never"
+    type      = "RESPONSE"
+    statement = "req.http.Fastly-Client-IP == \"127.0.0.1\" && req.http.Fastly-Client-IP != \"127.0.0.1\""
+  }
+
+  snippet {
+    content  = <<-EOT
+      if (req.url.path ~ "^/release/") {
+        set req.enable_segmented_caching = true;
+      }
+    EOT
+    name     = "Enable segment caching for large files"
+    priority = 60
+    type     = "recv"
+  }
+
+  # Allow CORS GET requests.
+  header {
+    destination = "http.access-control-allow-origin"
+    type        = "cache"
```

Comment:
I'd suggest setting `type` to `response`, instead of `cache`. This way the CORS header will be set right at the point Fastly serves the object, instead of caching the header. Unless you have a specific use case for this?

(E.g., at time of any particular security incident, you would then be able to change this `source` value quickly and easily, instead of having to purge the cache and update the header values.)

Reply:
Put the type to `response`.